replica/table.cc

```diff
@@ -66,30 +66,25 @@ static seastar::metrics::label keyspace_label("ks");
 
 using namespace std::chrono_literals;
 
-sstables::generation_type table::make_new_generation(std::optional<sstables::generation_type> prev) {
-    auto prev_value = prev.value_or(sstables::generation_type(0)).value();
-    auto next_value = prev_value - prev_value % smp::count + smp::count + this_shard_id();
-    tlogger.trace("new_generation {} -> {}", prev_value, next_value);
-    return sstables::generation_type(next_value);
-}
-
 void table::update_sstables_known_generation(std::optional<sstables::generation_type> generation) {
     auto gen = generation.value_or(sstables::generation_type(0)).value();
-    auto normalized_generation = gen - gen % smp::count + this_shard_id();
-    if (!_sstable_generation || normalized_generation > _sstable_generation->value()) {
-        _sstable_generation.emplace(normalized_generation);
-        tlogger.debug("{}.{} updated highest known generation to {}", schema()->ks_name(), schema()->cf_name(), *_sstable_generation);
+    auto normalized_gen = gen - gen % smp::count + this_shard_id();
+    if (_sstable_generation_generator) {
```

Comment:
With "normalization" moved to sstables::sstable_generation_generator, we can get rid of the arithmetic above too. User of the generator doesn't have to know about this. That's implementation detail pushed upwards.